app/Repositories/Category/CategoryUserRepository.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+
+namespace App\Repositories\Category;
+
+use App\CategoryUser;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+
+class CategoryUserRepository implements CategoryUserRepositoryContract
+{
+    public function insert($categories, int $id): bool
+    {
+        $userCategory = $this->userCategory($categories, $id);
+        return CategoryUser::insert($userCategory);
+    }
+
+    public function byUserId(int $id): Collection
+    {
+        return CategoryUser::where('user_id', $id)
+            ->get();
+    }
+
+    public function delete($id): bool
+    {
+        CategoryUser::where('user_id', $id)
+            ->delete();
+        return true;
+    }
+
+    public function store(array $categories)
+    {
+        if (isset($categories)) {
```

Comment:
Da der Parameter einen Typecast hat, ist der immer gesetzt. Du kannst aber folgendes fragen:
```suggestion
        if (count($categories)) {
```